### PR TITLE
Issue #1957 - Fix forwarding stdin to pre/post-receive scripts

### LINKED ISSUE #1957: No information is given on stdin for pre/post-receive
- Gitea version (or commit ref): 1.1.1
- Git version: 2.13.0.windows.1
- Operating system: Windows Server 2012 R2
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

[pre-receive](https://git-scm.com/docs/githooks#pre-receive) and [post-receive](https://git-scm.com/docs/githooks#post-receive) git hooks are supposed to receive information on `stdin`

related #1808

### PR DESCRIPTION
**Pull request by olivier-mauras**
Fix https://github.com/go-gitea/gitea/issues/1957
Ensure stdin is forwarded for pre/post-receive hooks

### CODE REVIEW

**Review by lafriks — CHANGES_REQUESTED** [review]
It should be as new migration step as otherwise scripts won't be updated for anyone using 1.1+ version

**Comment by olivier-mauras** [comment]
Oh now I have to get a grasp of those migration steps ^^

**Comment by lunny** [comment]
And maybe some integration test? There are many examples in `integrations` directory.

File: models/migrations/v22.go [hunk 1]
```diff
@@ -31,9 +31,9 @@ func generateAndMigrateWikiGitHooks(x *xorm.Engine) (err error) {
 	var (
 		hookNames = []string{"pre-receive", "update", "post-receive"}
 		hookTpls  = []string{
-			fmt.Sprintf("#!/usr/bin/env %s\nORI_DIR=`pwd`\nSHELL_FOLDER=$(cd \"$(dirname \"$0\")\";pwd)\ncd \"$ORI_DIR\"\nfor i in `ls \"$SHELL_FOLDER/pre-receive.d\"`; do\n    sh \"$SHELL_FOLDER/pre-receive.d/$i\"\ndone", setting.ScriptType),
+			fmt.Sprintf("#!/usr/bin/env %s\nORI_DIR=`pwd`\nSHELL_FOLDER=$(cd \"$(dirname \"$0\")\";pwd)\nread STDIN\ncd \"$ORI_DIR\"\nfor i in `ls \"$SHELL_FOLDER/pre-receive.d\"`; do\n    echo \"$STDIN\" | sh \"$SHELL_FOLDER/pre-receive.d/$i\"\ndone", setting.ScriptType),
```
Comment (bkcsoft) [inline models/migrations/v22.go:34]: No, this need a _new_ migration since this has already been ran...

**Review by bkcsoft — CHANGES_REQUESTED** [review]

File: models/migrations/v22.go (hunk 1 above)
Comment (olivier-mauras) [reply models/migrations/v22.go:34]: I agree for already migrated instances, but not on instances still on gogs that will migrate to gitea...
Or you mean that it's pointless anyway cause it needs its own migration step anyway that will happen in both cases ?

Comment (bkcsoft) [reply models/migrations/v22.go:34]: For instances migrating from Gogs, they will have to go through all migrations. The latest migration to touch a file "wins" 🙂

Comment (olivier-mauras) [reply models/migrations/v22.go:34]: Understood. I'll remove the modification from there and add it in its own migration step then.

**Comment by lafriks** [comment]
This issue is actually already resolved in commit https://github.com/go-gitea/gitea/commit/fe94032f74298569512f6ece2fbe4f7ab9a24e46 and this PR just fixes older migrations that will be overridden by migration v26 anyway

**Pull request closed without merging**